### PR TITLE
Add e2e coverage deps and scripts

### PR DESCRIPTION
**Pull request by ravichdev**
## Summary

Fixes #

## Checklist

- [ ] My pull request is addressing an [open issue](https://github.com/xwp/wp-foo-bar/issues) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/xwp/wp-foo-bar/contributing.md#scripts).
- [ ] My code follows the [Contributing Guidelines](https://github.com/xwp/wp-foo-bar/contributing.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -122,12 +127,14 @@
...
     "test:e2e:ci": "npm run test:e2e -- --runInBand",
     "test:e2e:coverage": "npm run test:e2e -- --coverage --coverageDirectory=tests/coverage/e2e",
-    "test:e2e:coveralls": "npm run test:e2e -- --coverage --coverageReporters=text-lcov | coveralls",
+    "test:e2e:coveralls": "npm run test:e2e:coverage -- --runInBand --coverageReporters=json && npm run test:js:coverage-merge -- --no-js --reporter=text-lcov | coveralls",
     "test:js": "wp-scripts test-unit-js tests/js",
     "test:js:help": "wp-scripts test-unit-js --help",
     "test:js:watch": "npm run test:js -- --watch",
     "test:js:coverage": "wp-scripts test-unit-js --coverage --coverageDirectory=tests/coverage/js",
-    "test:js:coveralls": "wp-scripts test-unit-js --coverage --coverageReporters=text-lcov | coveralls",
+    "test:js:coveralls": "npm run test:js:coverage && coveralls < tests/coverage/js/lcov.info",
```
Comment (ravichdev) [inline package.json:137]: There seems to be an issue with piping the coverage output to the `coveralls` command, failed test cases seem to pass the build. Eg (https://travis-ci.com/github/xwp/material-theme-builder-wp/jobs/317163271)

**Comment by derekherman** [comment]
@ravichdev I tested this locally and excluded the JS tests from the coverage report to see what the results of the e2e tests were and there was no coverage. It doesn't appear to actually add e2e test coverage. What happens is that the e2e tests don't cover any files so there are 0 out of 0 lines and the JS tests cover 100% of the JS code in `assets/src` so when you merge the two you get 100% coverage. Could you confirm that this does generate e2e coverage for you without any JS tests being ran? Also, there are conflicts that need to be resolved since there was a  bunch of packages updated and merged recently. Thank you for taking a crack at it, hopefully we can figure this out.

**Comment by ravichdev** [comment]
@derekherman How are you excluding only the JS tests ? I am seeing different results. I removed the `tests/js/block-editor/blocks/hello-world/index.spec.js` and `tests/js/block-editor/helpers/index.spec.as` files which have the JS coverage and then ran `npm run test:js:coverage-all`. Below are the results

**E2E**
e2e

**JS**
js

**JS & E2E**
e2e-js

Please note the `test:js:coverage-merge` command which does the JS and E2E merge is only for local development and it will not be used in Travis/Coveralls as Coveralls does it own merge and we send the JS and E2E coverage as their own own requests.

The E2E test also need the node env to be either `teste2e` or `development`

**Comment by derekherman** [comment]
@ravichdev you can run the following command to see exactly what I'm talking about.

```
NODE_ENV=teste2e; npm run test:e2e:coverage
```

E2E Coverage

**Comment by derekherman** [comment]
If you comment out line:56 in the `merge-coverage.js` file `reports.push( rootDir + '/js/coverage-final.json' );` and run `npm run test:js:coverage-all` then you will also see that e2e coverage is zero. And the `coverage-puppeteer-istanbul.json` file is an empty object.

**Comment by ravichdev** [comment]
I tried both and still seeing coverage:

```
→ NODE_ENV=teste2e; npm run test:e2e:coverage

> wp-foo-bar@ test:e2e:coverage /Users/ravi/Projects/xwp/wp-foo-bar
> npm run test:e2e -- --coverage --coverageDirectory=tests/coverage/e2e


> wp-foo-bar@ test:e2e /Users/ravi/Projects/xwp/wp-foo-bar
> WP_BASE_URL=http://localhost:8088 wp-scripts test-e2e --config=tests/e2e/jest.config.js "--coverage" "--coverageDirectory=tests/coverage/e2e"

 PASS  tests/e2e/specs/block-editor/blocks/hello-world/index.spec.js (9.11s)
  blocks: foo-bar/hello-world
    ✓ should be inserted (7229ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        9.716s
Ran all test suites.
---------------------------------|---------|----------|---------|---------|-------------------
File                             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------------------------|---------|----------|---------|---------|-------------------
All files                        |   69.23 |      100 |   42.86 |   72.73 |
 block-editor/blocks/hello-world |   83.33 |      100 |      50 |     100 |
  index.js                       |   83.33 |      100 |      50 |     100 |
 block-editor/helpers            |     100 |      100 |     100 |     100 |
  index.js                       |     100 |      100 |     100 |     100 |
 classic-editor                  |       0 |      100 |       0 |       0 |
  index.js                       |       0 |      100 |       0 |       0 | 2
 customizer                      |       0 |      100 |       0 |       0 |
  customize-controls.js          |       0 |      100 |       0 |       0 | 2
  customize-preview.js           |       0 |      100 |       0 |       0 | 2
 front-end                       |       0 |        0 |       0 |       0 |
  index.js                       |       0 |        0 |       0 |       0 |
---------------------------------|---------|----------|---------|---------|-------------------
```

I think the issue is the compiled/built files do not have the additional syntax the `babel-plugin-istanbul` plugin adds to detect coverage. Can you run `npm run dev` and then `npm run test:e2e:coverage`. `npm run build:js` is using the env as production and hence not generating the additional syntax required to detect puppeteer coverage.

**Pull request merged**